### PR TITLE
chore(claude-code): add hook guarding npm run e2e-test against missing .only

### PR DESCRIPTION
**Pull request by davidfirst**
Adds a `PreToolUse(Bash)` hook in `.claude/settings.json` that blocks `npm run e2e-test[:debug]` when no `.only` marker has been added in `e2e/` (checked via `git diff HEAD`). Without `.only` the full e2e suite runs for hours — exactly the failure mode CLAUDE.md warns about.

The check looks at *added* diff lines (not raw grep) because some e2e files embed `it.only(...)` inside fixture string literals (e.g. `e2e/harmony/mocha-tester.e2e.ts`). Quoted substrings in the bash command are stripped before matching so commit messages / echo args containing the phrase don't false-positive.

Also opens `.gitignore` exceptions for `.claude/settings.json` and `.claude/hooks/` so the guard ships with the repo; `.claude/settings.local.json` stays ignored.

### CODE REVIEW

**Review by GiladShoham — APPROVED** [review]

File: .claude/settings.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "$schema": "https://json.schemastore.org/claude-code-settings.json",
+  "hooks": {
+    "PreToolUse": [
+      {
+        "matcher": "Bash",
+        "hooks": [
+          {
+            "type": "command",
+            "command": "\"${CLAUDE_PROJECT_DIR}/.claude/hooks/check-e2e-only.sh\""
```
[the remark this answers is not in the document]
Comment (davidfirst) [reply .claude/settings.json]: Good nit — simplified in c34f1ad91. The escaped quotes were defensive against `$CLAUDE_PROJECT_DIR` containing spaces, but in practice it never does on a dev machine and the unquoted form matches the canonical hook examples. Verified live that the hook still fires.

File: .claude/hooks/check-e2e-only.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+# PreToolUse(Bash) guard: blocks `npm run e2e-test[:debug]` when no `.only`
+# marker has been added in e2e/. Without `.only` the full e2e suite runs for
+# hours. See CLAUDE.md ("Running specific e2e tests").
+set -uo pipefail
+
+INPUT=$(cat)
+CMD=$(printf '%s' "$INPUT" | jq -r '.tool_input.command // ""')
+
```
[the remark this answers is not in the document]
Comment (davidfirst) [reply .claude/hooks/check-e2e-only.sh:23]: Valid concern — fixed in c34f1ad91. The hook now fails closed on two failure modes: (1) `jq` missing on PATH → deny with a clear reason; (2) malformed JSON on stdin → deny with a clear reason. Verified with `echo not-json | check-e2e-only.sh` which now returns a `deny` permissionDecision instead of silently allowing.

**Comment by davidfirst** [comment]
Responding to the two low-confidence comments in the Copilot review summary:

**1. `&&` / `||` chaining (line 20):** The regex `[;&|][[:space:]]*npm` already allows zero whitespace after the separator, so `&& npm` matches the second `&` followed by space. Verified live earlier this session:
```
echo '{"tool_name":"Bash","tool_input":{"command":"bit compile && npm run e2e-test"}}' | .claude/hooks/check-e2e-only.sh
→ deny ✓
```
No code change.

**2. Untracked files (line 28):** Real edge case — fixed in c34f1ad91. CLAUDE.md tells Claude to create new e2e files for bug repros, so an untracked `.only`-containing test would have been wrongly denied. The hook now also scans `git ls-files --others --exclude-standard -- e2e/` for untracked files containing `.only`. Updated the comment to match. Verified with a temp untracked file containing `it.only` — hook allows.
